sdk/search/search/samples/typescript/src/readonlyQuery.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * Performs a query over a public dataset
+ */
+
+import { SearchIndexClient, SearchApiKeyCredential, odata, GeographyPoint } from "@azure/search";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+interface Hotel {
+  HotelId: string;
```

Comment:
Are all of these types potentially nullable in Typescript? Asking because every field type is nullable except for collection types. Trying to deserialize null into a type that doesn't accept null is usually a bad thing.

Reply:
They can either make the fields optional or use the Partial mapped type. We deserialize it fine either way, I show it this way so you don't have to conditionally check each and every field before you dot into it... But maybe we should force it to a partial type just in case?

Reply:
I'm not sure what the best practice should be, being new to TypeScript. This is much easier in C# where all reference types are nullable by default anyway.

Reply:
So, you have option in TypeScript. If you're not in strict mode, everything can be `null` or `undefined`. In strict mode you can specify that manually e.g.

```ts
interface Hotel {
  HotelId: string;
  Description?: string;
  LastRenovationDate: Date | null;
}
```

Really this is a bit different than .NET and others because you're not specifying a type to deserialize into, you're just describing for the TS compiler what you expect at runtime, so the developer has complete control to write what is accurate/best for them.

The worst case scenario is the developer writes incorrect typings and then at runtime a field they try to access isn't the type the TS compiler expected and they get a runtime exception. This is true of using TS to type any kind of remote payload, though.

Reply:
If I understand correctly, this means we really don't need to do anything in code per se... just point out that any field can be null, and let developers use their preferred typing strategy.

Reply:
Yep! I will also keep an eye on this specifically when we have TS customers try out the library